third_party/mesa/src/compiler/isaspec/isaspec_decode_impl.c

```diff
@@ -733,7 +776,13 @@ display(struct decode_scope *scope)
 				e++;
 			}
 
-			char *field_name = strndup(p, e-p);
+#ifdef _MSC_VER
+            char *field_name = (char *)malloc(e - p + 1); 
+            strncpy(field_name, p, e - p);
+            field_name[e - p] = '\0'; 
+#else
+            char *field_name = strndup(p, e - p);
```

Comment:
out of curiosity, `strndup` is not working on windows?

Reply:
would it be cleaner if we encode our own implementation into a header/cpp and wrap the header with #ifdef _MSC_VER

something like:
```
google_dive_ext.h
namespace Dive
{
...
char *strndup(const char *s, size_t n)
...
}

google_dive_ext.cpp
char *strndup(const char *s, size_t n) {
    size_t len = strnlen(s, n); 
    char *new_str = (char *)malloc(len + 1);
    if (new_str == NULL) {
        return NULL;
    }
    memcpy(new_str, s, len);
    new_str[len] = '\0'; 
    return new_str;
}
```
in this case, we could isolate our changes from the other part of the mesa code, would be much easier for future integration